packages/cloud/src/queries/tenants.ts

```diff
@@ -44,30 +44,38 @@ export const createTenantsQueries = (client: Queryable<PostgreSql>) => {
     `);
 
   const insertAdminData = async (data: AdminData) => {
-    const { resource, scope, role } = data;
+    const { resource, scopes, role } = data;
 
     assert(
-      resource.tenantId && scope.tenantId && role.tenantId,
+      resource.tenantId && scopes.every(({ tenantId }) => tenantId) && role.tenantId,
       new Error('Tenant ID cannot be empty.')
     );
 
     assert(
-      resource.tenantId === scope.tenantId && scope.tenantId === role.tenantId,
+      scopes.every(
+        (scope) => resource.tenantId === scope.tenantId && scope.tenantId === role.tenantId
+      ),
       new Error('All data should have the same tenant ID.')
     );
 
     await client.query(insertInto(resource, 'resources'));
-    await client.query(insertInto(scope, 'scopes'));
+    await Promise.all(scopes.map(async (scope) => client.query(insertInto(scope, 'scopes'))));
     await client.query(insertInto(role, 'roles'));
-    await client.query(
-      insertInto(
-        {
-          id: generateStandardId(),
-          roleId: role.id,
-          scopeId: scope.id,
-          tenantId: resource.tenantId,
-        } satisfies CreateRolesScope,
-        'roles_scopes'
+
+    const { tenantId } = resource; // Linter failed to infer `tenantId` from previous assertion.
```

Comment:
- the error does not come from linter, it's TS type check
- the comment looks ambiguous and may cause confusion, and it's normal to use deconstructing here. so we can remove it.

```suggestion
    const { tenantId } = resource;
```